src/v/pandaproxy/schema_registry/types.h

```diff
@@ -59,21 +56,39 @@ from_string_view<schema_type>(std::string_view sv) {
       .default_match(std::nullopt);
 }
 
-inline std::ostream& operator<<(std::ostream& os, const schema_type& v) {
-    return os << to_string_view(v);
-}
+std::ostream& operator<<(std::ostream& os, const schema_type& v);
 
 ///\brief A subject is the name under which a schema is registered.
 ///
 /// Typically it will be "<topic>-key" or "<topic>-value".
 using subject = named_type<ss::sstring, struct subject_tag>;
 static const subject invalid_subject{};
 
+///\brief An unvalidated definition of the schema and its type.
+struct raw_schema_definition {
+    using str = named_type<ss::sstring, struct str_tag>;
```

Comment:
`str  and str_tag` are hard to understand what it means. both generic things. 

what kind of `str` is this.